### ISSUE TITLE
Define the configuration and how to send the foreign keys

### ISSUE THREAD

**Issue by tatsumibruno** [tecsinapse/tecsinapse-data-sync#13]
The goal of this issue is define the configuration that will be required in the entities to send/receive foreign keys fields.

A way to do this as below:
@SyncField(field="dealer", foreignKeyField="id", foreignKeyEntity="dealer", localField="dealer_id")

I think thats gonna work, but its a little complicated to configure.

What you think guys?

cc @filipenevola @leonardoterrao

**Comment by leonardoterrao** [comment]
Very good example @Tatsumi. It may be simpler for mapped classes where from the type we can get the information from the other entity. Have all of this data filled out only for entities annotated with @ElementCollection, where the type can be a map.

**Comment by filipenevola** [comment]
Exactly, and in fields that we don't know that information we can have a default, like id, entityName_id, etc and it only will be mandatory when you really need something different. And probably foreignKeyEntity should receive Dealer.class instead of a String

**Comment by leonardoterrao** [comment]
I tried to evolve with the relationship part. I tried alone and I didn't succeed. I was a bit confused about how to proceed after I got the name of the fields, how to build the relationship. I'm a little lost because I also thought about relationships annotated with @ManyToMany and @EmentmentCollection.

I think of a check list to follow, just to help focus:
- Add the remaining values for the @SyncField annotation that will be used in the relationship, fields that were suggested by @tatsumibruno
- In reading the classes, each class will capture the normal fields and the relationship fields
- Mount a new SQL builder to be prepared to capture the fields and fill in the joins when there are relationships.
- Create triggers for tables that are affected by the relationship (foreignKeyEntity).

**Comment by tatsumibruno** [comment]
@leonardoterrao i have some ideas, but did not tried yet, like @filipenevola said before, we can annotate the field with the target class, and that (foreign) class has to be annotated too, and woth that we can build the query. What you think?
